backend/app/history/chat_history.py

```diff
@@ -0,0 +1,84 @@
+"""
+This file contains all the logic for communicating with the database with regard to chat history messages.
+Chat messages are managed using a session key and set to expire after some time if not used.
+
+Author: Ruben Vandamme
+Date: 05 November 2024
+Usage:
+    - add_message           : save a message
+    - get_chat_history      : get messages using a session key
+    - (refresh_session)     : refresh messages TTL of a session
+    - find_session_id       : check if a session id is present
+    - generate_session_id   : generate an unique session id
+"""
+
+from pymongo import MongoClient
+from pymongo import ASCENDING
+import datetime
+from pydantic import BaseModel
+from typing import List
+from enum import StrEnum
+import secrets
+import string
+
+# setup connection
+client = MongoClient('mongodb://localhost:27017/')
+db = client['chat_db']
+chat_collection = db['chat_history']
+chat_collection.create_index([("created_at", ASCENDING)], expireAfterSeconds=3600)
+
+# models
+class UserOrChatbot(StrEnum):
+    USER                = "user"
+    CHATBOT             = "chatbot"
+class InsertMessage(BaseModel):
+    session_id          : str
+    user_or_chatbot     : UserOrChatbot
+    message             : str
+class RetrieveMessage(BaseModel):
+    session_id          : str
+    user_or_chatbot     : UserOrChatbot
+    message             : str
+    created_at          : datetime.datetime
+
+# interaction functions
+def add_message(message_info: InsertMessage):
+    refresh_session(message_info.session_id)
+    chat_message = {
+        "session_id"        : message_info.session_id,
+        "user_or_chatbot"   : message_info.user_or_chatbot,
+        "message"           : message_info.message,
+        "created_at"        : datetime.datetime.now(datetime.timezone.utc)
+    }
+    chat_collection.insert_one(chat_message)
+
+def get_chat_history(
+    session_id  : str, 
+    limit       : int   = 4     # choose even so that user query always comes first
+) -> List[RetrieveMessage]:
```

Comment:
you can use `list[RetrieveMessage]`. Lowercase doesn't require the import. We are using more modern python so this is preferable.